### ISSUE TITLE
(@aws-amplify/ui-react): Component re-render on I18n.setLanguage

### ISSUE THREAD

**Issue by wei** [aws-amplify/amplify-js#6986]
**Is your feature request related to a problem? Please describe.**
Components are not re-rendering on `I18n.setLanguage`. Components are rendering initial language strings correctly but they do not re-render on language change.
Relates to #6346 #6115 #6962

**Describe the solution you'd like**
Components should re-render on `I18n.setLanguage`

Suggestion from @ericclemmons: https://github.com/aws-amplify/amplify-js/pull/6346#issuecomment-710227501

**Describe alternatives you've considered**
1. Force re-render using `key` attribute
    ```jsx
    <AmplifySignIn key={lang} />
    ```
    Force re-creating the components will result in a screen flicker and loss of component state
2. Specifying all texts in props
    ```jsx
    <AmplifySignIn 
      headerText={I18n.get(Translations.XXXXX)}
      formFields={[
        {
          type: "email",
          label: I18n.get(Translations.XXXXX),
          placeholder: I18n.get(Translations.USERNAME_PLACEHOLDER),
          required: true,
        },
        {
          type: "password",
          label: I18n.get(Translations.XXXXX),
          placeholder: I18n.get(Translations.PASSWORD_PLACEHOLDER),
          required: true,
        }
      ]}>
    </AmplifySignIn>
    ```
    Doable but a big hassle : D

**Additional context**
Need to check if other ui-components e.g. angular/vue, etc exhibit the same behavior.

**Comment by princiya** [comment]
@wei is this issue related to this one? https://github.com/aws-amplify/amplify-js/issues/6898

**Comment by wei** [comment]
@princiya No, I think that's a different issue about Translations lacking some strings.

**Comment by mauerbac** [comment]
hey @ericclemmons - can you take a look at this ?

**Comment by ericclemmons** [comment]
I'd be interested in seeing how well https://github.com/aws-amplify/amplify-js/pull/6346#issuecomment-710227501 works, but the problem may be with Stencil's re-rendering potentially blowing away values in internal `state` or `<input>`s.

**Comment by mauerbac** [comment]
Let's discuss at an upcoming bug bash @sammartinez

**Comment by eokoneyo** [comment]
Hi there, I'm curious if there's any updates regarding this issue ... I seem to have run into the same issue

**Comment by wei** [comment]
@eokoneyo For now, you can specify attributes directly as seen in https://github.com/aws-amplify/amplify-js/issues/6962#issuecomment-710701413
